File: backend/app/models/visit_log.py
"""Visit log ORM model."""
from __future__ import annotations

import uuid
from datetime import datetime
from typing import Optional

from sqlalchemy import DateTime, Index, String, func
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class VisitLog(Base):
    __tablename__ = "visit_logs"

    id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        primary_key=True,
        default=uuid.uuid4,
        server_default=func.gen_random_uuid(),
    )
    ip: Mapped[str] = mapped_column(String(64), index=True, nullable=False)
    user_agent: Mapped[str] = mapped_column(String(512), default="", nullable=False)
    path: Mapped[str] = mapped_column(String(256), nullable=False)
    referer: Mapped[Optional[str]] = mapped_column(String(512), nullable=True)
    user_id: Mapped[Optional[uuid.UUID]] = mapped_column(UUID(as_uuid=True), nullable=True)
    visited_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )

    __table_args__ = (
        Index("ix_visit_logs_visited_at", "visited_at"),
    )

File: backend-java/src/main/java/com/myblog/application/model/entity/VisitLog.java
package com.myblog.application.model.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.OffsetDateTime;
import java.util.UUID;

@Data
@TableName("visit_logs")
public class VisitLog {

    @TableId(type = IdType.ASSIGN_UUID)
    private UUID id;

    private String ip;
    private String userAgent;
    private String path;
    private String referer;
    private UUID userId;
    private OffsetDateTime visitedAt;
}
